Added continuous consumer group mirroring test

mmaslankaprv opened the pull request:
Added a test verifying if consumer group offsets are continuously mirrored from source to the target cluster.

<!--
See https://github.com/redpanda-data/redpanda/blob/dev/CONTRIBUTING.md#pull-request-body
for more details and examples of what is expected in a PR body.

Content in this top section is REQUIRED. Describe, in plain language, the motivation
behind the change (bug fix, feature, improvement) in this PR and how the included
commits address it.

Add the GitHub keyword `Fixes` to link to bug(s) this PR will fix, e.g.
  Fixes #ISSUE-NUMBER, Fixes #ISSUE-NUMBER, ...

If this PR is a backport, link to the original with `Backport of PR`, e.g.
  Backport of PR #PR-NUMBER
-->

## Backports Required

<!-- Checking at least one of the checkboxes is REQUIRED if this PR is not a backport. -->

- [ ] none - not a bug fix
- [ ] none - this is a backport
- [ ] none - issue does not exist in previous branches
- [ ] none - papercut/not impactful enough to backport
- [ ] v25.2.x
- [ ] v25.1.x
- [ ] v24.3.x

## Release Notes

<!--
If the changes in this PR do not need to be mentioned in the release
notes, then don't add a sub-section and simply list `none`, e.g.

* none

Otherwise, adding a sub-section or `none` is REQUIRED if the PR is not a backport PR.
If this is a backport PR, adding contents to this section will override
the release notes section inherited from the original PR to dev.

Add one or more of the sub-sections with a short description bullet
point of the change, e.g.

### Bug Fixes

* Short description of the bug fix if this is a PR to `dev` branch.

### Features

* Short description of the feature. Explain how to configure.

### Improvements

* Short description of how this PR improves existing behavior.

-->
- none

bharathv commented on tests/rptest/tests/cluster_linking_e2e_test.py:
```diff
@@ -245,44 +285,91 @@ def _group_present_in_target_cluster():
-            cleanup_policy = "delete" if i % 2 == 0 else "compact"
-            topic = TopicSpec(
+    @cluster(num_nodes=7)
+    @matrix(with_failures=[False])
```
=[True, False] ?

mmaslankaprv replied:
not yet :-) as we have the assert failing, waiting for it to be fixed

bharathv replied:
Ah, this has regular topics too

bharathv commented on tests/rptest/tests/cluster_linking_e2e_test.py:
```diff
@@ -187,6 +185,50 @@ def test_can_not_create_more_than_one_link(self):
+                return False
+            for t in topics:
+                if t.name not in topics_in_target:
+                    return False
```
I factored a method into the base class

topic_present_in_target(topic: list|topic) -> bool .. seems like a useful utility being used all over.

vbotbuildovich commented:
<!--TEST RESULTS-->
<!--1adbe769-cf34-404b-a450-0c8ecb4f1bbf-->
<!--TEST RESULTS-->
<!--93f4c474-a6ca-47e6-8793-85aea4709a71-->
<!--TEST RESULTS-->
<!--fbd0170c-94aa-4a37-a2f1-c2d27753bddb-->
<!--TEST RESULTS-->
<!--c17d59fd-7651-4428-aa33-35e020159125-->
<!--title-->
## CI test results
<!--/title-->
<!--body-->
<details>
<summary>test results on <a href="https://buildkite.com/redpanda/redpanda/builds/72081">build#72081</a></summary>


| test_class         | test_method            | test_arguments           | test_kind   | job_url                                                                                   | test_status   | passed   | reason                                                                                                                                                                 |
|:-------------------|:-----------------------|:-------------------------|:------------|:------------------------------------------------------------------------------------------|:--------------|:---------|:-----------------------------------------------------------------------------------------------------------------------------------------------------------------------|
| ShardPlacementTest | test_core_count_change | null                     | integration | https://buildkite.com/redpanda/redpanda/builds/72081#0199390b-a92e-43b8-a149-65b1d9789f5a | FLAKY         | 20/21    | upstream reliability is '99.64912280701755'. current run reliability is '95.23809523809523'. drift is 4.41103 and the allowed drift is set to 50. The test should PASS |
| SimpleEndToEndTest | test_relaxed_acks      | {"write_caching": false} | integration | https://buildkite.com/redpanda/redpanda/builds/72081#0199390b-a931-4de8-80b4-183179493f00 | FLAKY         | 20/21    | upstream reliability is '100.0'. current run reliability is '95.23809523809523'. drift is 4.7619 and the allowed drift is set to 50. The test should PASS              |
</details>
<details>
<summary>test results on <a href="https://buildkite.com/redpanda/redpanda/builds/72140">build#72140</a></summary>


| test_class                           | test_method                         | test_arguments                                                                                                                        | test_kind   | job_url                                                                                   | test_status   | passed   | reason                                                                                                                                                                   |
|:-------------------------------------|:------------------------------------|:--------------------------------------------------------------------------------------------------------------------------------------|:------------|:------------------------------------------------------------------------------------------|:--------------|:---------|:-------------------------------------------------------------------------------------------------------------------------------------------------------------------------|
| AuditLogTestOauth                    | test_kafka_oauth                    | {"authz_match": "rbac"}                                                                                                               | integration | https://buildkite.com/redpanda/redpanda/builds/72140#01993dbb-fc6d-41c2-94ca-1b1450e2579b | FLAKY         | 13/21    | upstream reliability is '91.26984126984127'. current run reliability is '61.904761904761905'. drift is 29.36508 and the allowed drift is set to 50. The test should PASS |
| ShadowLinkConsumeGroupsMirroringTest | test_continuous_group_sync          | {"with_failures": false}                                                                                                              | integration | https://buildkite.com/redpanda/redpanda/builds/72140#01993dbb-fc68-4dc3-869b-77d1e889b0e9 | FLAKY         | 19/21    | The test was found to be new, and no failures are allowed                                                                                                                |
| ShadowLinkingReplicationTests        | test_replication_with_failures      | null                                                                                                                                  | integration | https://buildkite.com/redpanda/redpanda/builds/72140#01993db4-b97e-4bc3-810a-245ef75b4ac5 | FLAKY         | 20/21    | upstream reliability is '100.0'. current run reliability is '95.23809523809523'. drift is 4.7619 and the allowed drift is set to 50. The test should PASS                |
| SchemaEvolutionE2ETests              | test_legal_schema_evolution         | {"catalog_type": "rest_hadoop", "cloud_storage_type": 1, "produce_mode": "avro", "query_engine": "trino", "test_case": "drop_column"} | integration | https://buildkite.com/redpanda/redpanda/builds/72140#01993dbb-fc66-4ce0-81a3-0b2caaf2f71b | FLAKY         | 20/21    | upstream reliability is '100.0'. current run reliability is '95.23809523809523'. drift is 4.7619 and the allowed drift is set to 50. The test should PASS                |
| SchemaEvolutionE2ETests              | test_reorder_columns                | {"catalog_type": "rest_hadoop", "cloud_storage_type": 1, "produce_mode": "avro", "query_engine": "trino"}                             | integration | https://buildkite.com/redpanda/redpanda/builds/72140#01993dbb-fc70-49bd-aff8-9cb8e7e6f1d4 | FLAKY         | 20/21    | upstream reliability is '99.3421052631579'. current run reliability is '95.23809523809523'. drift is 4.10401 and the allowed drift is set to 50. The test should PASS    |
| PartitionBalancerTest                | test_recovery_mode_rebalance_finish | null                                                                                                                                  | integration | https://buildkite.com/redpanda/redpanda/builds/72140#01993db4-b97b-43e1-a253-eba63b7261cc | FLAKY         | 12/21    | upstream reliability is '94.89795918367348'. current run reliability is '57.14285714285714'. drift is 37.7551 and the allowed drift is set to 50. The test should PASS   |
</details>
<details>
<summary>test results on <a href="https://buildkite.com/redpanda/redpanda/builds/72171">build#72171</a></summary>


| test_class                           | test_method                          | test_arguments          | test_kind   | job_url                                                                                   | test_status   | passed   | reason                                                                                                                                                                 |
|:-------------------------------------|:-------------------------------------|:------------------------|:------------|:------------------------------------------------------------------------------------------|:--------------|:---------|:-----------------------------------------------------------------------------------------------------------------------------------------------------------------------|
| ShadowLinkConsumeGroupsMirroringTest | test_continuous_group_sync           | {"with_failures": true} | integration | https://buildkite.com/redpanda/redpanda/builds/72171#01993ec1-09fd-4550-97ce-5ce824260956 | FAIL          | 0/1      |                                                                                                                                                                        |
| DataMigrationsApiTest                | test_creating_and_listing_migrations | null                    | integration | https://buildkite.com/redpanda/redpanda/builds/72171#01993ec1-0a02-4199-83b1-cd29c8d5ef0a | FLAKY         | 20/21    | upstream reliability is '99.71428571428571'. current run reliability is '95.23809523809523'. drift is 4.47619 and the allowed drift is set to 50. The test should PASS |
</details>
<details>
<summary>test results on <a href="https://buildkite.com/redpanda/redpanda/builds/72354">build#72354</a></summary>


| test_class                 | test_method            | test_arguments                                                                                                                       | test_kind   | job_url                                                                                   | test_status   | passed   | reason                                                                                                                                                              |
|:---------------------------|:-----------------------|:-------------------------------------------------------------------------------------------------------------------------------------|:------------|:------------------------------------------------------------------------------------------|:--------------|:---------|:--------------------------------------------------------------------------------------------------------------------------------------------------------------------|
| DatalakeClusterRestoreTest | test_basic             | {"catalog_type": "rest_jdbc", "cloud_storage_type": 1}                                                                               | integration | https://buildkite.com/redpanda/redpanda/builds/72354#0199535c-feee-4e6a-82f1-578e78f6aed4 | FLAKY         | 20/21    | upstream reliability is '100.0'. current run reliability is '95.23809523809523'. drift is 4.7619 and the allowed drift is set to 50. The test should PASS           |
| SchemaEvolutionE2ETests    | test_old_schema_writer | {"catalog_type": "rest_hadoop", "cloud_storage_type": 1, "produce_mode": "avro", "query_engine": "trino", "test_case": "add_column"} | integration | https://buildkite.com/redpanda/redpanda/builds/72354#01995356-7553-4fcb-8b84-c8cbda4ff3ac | FLAKY         | 20/21    | upstream reliability is '99.3920972644377'. current run reliability is '95.23809523809523'. drift is 4.154 and the allowed drift is set to 50. The test should PASS |
</details>
<!--/body-->

mmaslankaprv committed:
cl/tests: added continuous consumer group mirroring test

Added a test verifying mirroring of consumer groups when they are
actively updated on the source cluster.

Signed-off-by: Michał Maślanka <michal@redpanda.com>

vbotbuildovich commented:
<!--FAILURES SLASH COMMAND https://buildkite.com/redpanda/redpanda/builds/72140-->
<!--2b1469b4-a446-413b-8dba-0e9fd4b250f3-->
<!--title-->
## Retry command for [Build#72140](https://buildkite.com/redpanda/redpanda/builds/72140)
please wait until all jobs are finished before running the slash command
<!--/title-->
<!--body-->
```
/ci-repeat 1
tests/rptest/tests/cluster_linking_e2e_test.py::ShadowLinkConsumeGroupsMirroringTest.test_continuous_group_sync@{"with_failures":false}
```
<!--/body-->

vbotbuildovich commented:
<!--FAILURES SLASH COMMAND https://buildkite.com/redpanda/redpanda/builds/72171-->
<!--a8a72971-6f22-4cb3-a499-5d06857b8f21-->
<!--title-->
## Retry command for [Build#72171](https://buildkite.com/redpanda/redpanda/builds/72171)
please wait until all jobs are finished before running the slash command
<!--/title-->
<!--body-->
```
/ci-repeat 1
tests/rptest/tests/cluster_linking_e2e_test.py::ShadowLinkConsumeGroupsMirroringTest.test_continuous_group_sync@{"with_failures":true}
```
<!--/body-->

bharathv commented:
There seems to be a bug that results in non monotonic fetches that is causing this test to fail, reproduces locally, better not to merge until that is fixed.

bharathv commented:
> There seems to be a bug that results in non monotonic fetches that is causing this test to fail, reproduces locally, better not to merge until that is fixed.

this seems to have been fixed by https://github.com/redpanda-data/redpanda/pull/27546/commits/f7cc8eb5bbbcca70fd0bab5e27c1c6fff18c794e but the test is still a bit flaky.

bharathv commented:
/dt

mmaslankaprv commented:
/ci-repeat 4
release
dt-repeat 5
skip-redpanda-build
tests/rptest/tests/cluster_linking_e2e_test.py::ShadowLinkConsumeGroupsMirroringTest

mmaslankaprv commented:
/ci-repeat 4
release
dt-repeat=5
skip-redpanda-build
tests/rptest/tests/cluster_linking_e2e_test.py::ShadowLinkConsumeGroupsMirroringTest

vbotbuildovich commented:
<!--FAILURES SLASH COMMAND https://buildkite.com/redpanda/redpanda/builds/72350-->
<!--8389cd76-273f-42ca-931c-148b8927d025-->
<!--title-->
## Retry command for [Build#72350](https://buildkite.com/redpanda/redpanda/builds/72350)
please wait until all jobs are finished before running the slash command
<!--/title-->
<!--body-->
```
/ci-repeat 1
tests/rptest/tests/cluster_linking_e2e_test.py::ShadowLinkConsumeGroupsMirroringTest.test_continuous_group_sync@{"with_failures":true}
```
<!--/body-->
